### PR TITLE
Do not leak child processes when cancelling

### PR DESCRIPTION
**Pull request by hughsie**
See https://github.com/fwupd/fwupd/pull/4372

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [X] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by djcampello** [comment]
This failed for me:

```
Bail out! ERROR:../fwupd-9999/libfwupdplugin/fu-self-test.c:1052:fu_common_spawn_timeout_func: assertion failed (error == (g-io-error-quark, 19)): error is NULL
stderr:
**
ERROR:../fwupd-9999/libfwupdplugin/fu-self-test.c:1052:fu_common_spawn_timeout_func: assertion failed (error == (g-io-error-quark, 19)): error is NULL
```

**Comment by djcampello** [comment]
Just noticed the same in the CI :)

**Comment by hughsie** [comment]
Urgh, and me also. ETOOLATE for me -- ideas welcome.

File: libfwupdplugin/fu-common.c [hunk 1]
```diff
@@ -838,10 +838,14 @@ fu_common_spawn_sync(const gchar *const *argv,
 	fu_common_spawn_create_pollable_source(helper);
 	g_main_loop_run(helper->loop);
 	g_cancellable_disconnect(cancellable, cancellable_id);
-	if (g_cancellable_set_error_if_cancelled(helper->cancellable, error))
-		return FALSE;
 #endif
-	return g_subprocess_wait_check(subprocess, cancellable, error);
+	if (!g_subprocess_wait_check(subprocess, cancellable, error))
```
Comment (djcampello) [inline libfwupdplugin/fu-common.c:842]: helper->cancellable

File: libfwupdplugin/fu-common.c [hunk 2]
```diff
@@ -838,10 +838,14 @@ fu_common_spawn_sync(const gchar *const *argv,
...
+		return FALSE;
+	if (g_cancellable_set_error_if_cancelled(cancellable, error))
```
Comment (djcampello) [inline libfwupdplugin/fu-common.c:844]: helper->cancellable

Comment (hughsie) [reply libfwupdplugin/fu-common.c]: Correct -- I told you it was late! Can you try what I just pushed please.

Comment (djcampello) [reply libfwupdplugin/fu-common.c]: The test does not leak processes anymore! Thanks!

**Pull request merged**